Autogenerated application with `oidc-db-token-state-manager` fails to be built

### Describe the bug

I use `create` option of `quarkus-maven-plugin` to generate an app with `oidc-db-token-state-manager` extension. When I try to build the app (without any changes) it fails.

### Expected behavior

The app is built

### Actual behavior

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.8.1:build (default) on project reproducer-oidc: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.oidc.db.token.state.manager.OidcDbTokenStateManagerProcessor#produceDbTokenStateManagerBean threw an exception: java.lang.NullPointerException: Cannot read field "reactiveClient" because "sqlClientBuildItem" is null
[ERROR] 	at io.quarkus.oidc.db.token.state.manager.OidcDbTokenStateManagerProcessor.produceDbTokenStateManagerBean(OidcDbTokenStateManagerProcessor.java:47)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] 
[ERROR] 	[error]: Build step io.quarkus.oidc.db.token.state.manager.OidcDbTokenStateManagerProcessor#createDbTokenStateInitializerProps threw an exception: java.lang.NullPointerException: Cannot read field "reactiveClient" because "sqlClientBuildItem" is null
[ERROR] 	at io.quarkus.oidc.db.token.state.manager.OidcDbTokenStateManagerProcessor.createDbTokenStateInitializerProps(OidcDbTokenStateManagerProcessor.java:115)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]

```

### How to Reproduce?

1. Generate the app: `mvn -Dquarkus-plugin.version=3.8.1 -DquarkusRegistryClient=false io.quarkus:quarkus-maven-plugin:3.8.1:create -DprojectGroupId=io.quarkus.qe -DprojectVersion=1.0.0-SNAPSHOT -Dextensions=oidc-db-token-state-manager -DprojectArtifactId=reproducer`
2. `cd reproducer`
3. Build the app: `mvn clean package -DskipTests`

### Output of `uname -a` or `ver`

6.7.3-200.fc39.x86_64

### Output of `java -version`

Java version: 21.0.1, vendor: Eclipse Adoptium

### Quarkus version or git rev

3.8.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

Same error for 3.8.2

--- a/extensions/oidc-db-token-state-manager/deployment/src/main/java/io/quarkus/oidc/db/token/state/manager/OidcDbTokenStateManagerProcessor.java
+++ b/extensions/oidc-db-token-state-manager/deployment/src/main/java/io/quarkus/oidc/db/token/state/manager/OidcDbTokenStateManagerProcessor.java
@@ -18,10 +18,8 @@ import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
 import io.quarkus.arc.deployment.BeanContainerBuildItem;
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
 import io.quarkus.arc.deployment.SyntheticBeansRuntimeInitBuildItem;
-import io.quarkus.arc.deployment.ValidationPhaseBuildItem;
 import io.quarkus.builder.item.SimpleBuildItem;
 import io.quarkus.deployment.Capabilities;
-import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.BuildSteps;
 import io.quarkus.deployment.annotations.Consume;
@@ -78,7 +76,6 @@ public class OidcDbTokenStateManagerProcessor {
 
     @BuildStep
     ReactiveSqlClientBuildItem validateReactiveSqlClient(
-            BuildProducer<ValidationPhaseBuildItem.ValidationErrorBuildItem> validationErrors,
             Capabilities capabilities) {
         ReactiveSqlClientBuildItem sqlClientDbTable = null;
         for (String reactiveClient : SUPPORTED_REACTIVE_CLIENTS) {
@@ -86,17 +83,15 @@ public class OidcDbTokenStateManagerProcessor {
                 if (sqlClientDbTable == null) {
                     sqlClientDbTable = new ReactiveSqlClientBuildItem(reactiveClient);
                 } else {
-                    validationErrors.produce(new ValidationPhaseBuildItem.ValidationErrorBuildItem(
-                            new ConfigurationException("The OpenID Connect Database Token State Manager extension is "
-                                    + "only supported when exactly one Reactive SQL Client extension is present.")));
-                    return null;
+                    throw new ConfigurationException("The OpenID Connect Database Token State Manager extension is "
+                            + "only supported when exactly one Reactive SQL Client extension is present.");
                 }
             }
         }
         if (sqlClientDbTable == null) {
-            validationErrors.produce(new ValidationPhaseBuildItem.ValidationErrorBuildItem(new ConfigurationException(
+            throw new ConfigurationException(
                     "The OpenID Connect Database Token State Manager extension requires Reactive SQL Client extension. "
-                            + "Please refer to the https://quarkus.io/guides/reactive-sql-clients for more information.")));
+                            + "Please refer to the https://quarkus.io/guides/reactive-sql-clients for more information.");
         }
         return sqlClientDbTable;
     }

